### PR TITLE
Upgrade to actix 1.0

### LINKED ISSUE #6620: Upgrade to actix 1.0
Upgrade to actix 1.0.

A first attempt at the upgrade can be found [here](https://github.com/habitat-sh/habitat/tree/dmcneil/upgrade-actix-web).

The two main blockers are the `Middleware` and `Predicate` traits. The way middleware is handled has been completely changed. There are [examples](https://github.com/actix/examples/blob/master/middleware/src/simple.rs) of the new system. However, there seems to be many questions by the community on how to best implement middlware so I am going to hold off and see how the actix-web team responds. I can not find any equivalent of `Predicate` or filtering requests in the 1.0 release. I asked a question on their Gitter channel and am waiting to hear back.

Note: Also upgrade builder.

### PR DESCRIPTION
**Pull request by davidMcneil**
Resolves #6620

Signed-off-by: David McNeil <mcneil.david2@gmail.com>

### CODE REVIEW

File: components/sup/src/http_gateway.rs [hunk 1]
```diff
@@ -239,55 +264,42 @@ impl Server {
...
                     ServerStartup::BindFailed
                 }
             };
 
             cvar.notify_one();
 
-            sys.run();
+            if let Ok(b) = bind {
+                b.run().expect("to start http server");
+            }
```
Comment (raskchanky) [inline components/sup/src/http_gateway.rs]: Why move this block outside the lock `match` on line 266?  Why not just change `b.start();` to `b.run().expect("to start http server");`?

Comment (raskchanky) [reply components/sup/src/http_gateway.rs]: Ah, unless `b.run()` is the point of no return, like `sys.run()` was before?

Comment (davidMcneil) [reply components/sup/src/http_gateway.rs]: Correct. That being said I am not a hug fan of matching twice on bind so if you have a better way to formulate it that would be awesome.

File: components/sup/src/http_gateway.rs [hunk 2]
```diff
@@ -112,71 +115,93 @@ impl AppState {
...
+
+    srv.call(req).and_then(|res| {
+                     let timer = res.request()
+                                    .app_data::<AppState>()
+                                    .expect("app data")
+                                    .timer
+                                    .replace(None);
+                     if timer.is_some() {
+                         timer.unwrap().observe_duration();
+                     }
```
Comment (baumanj) [inline components/sup/src/http_gateway.rs]: I realize this is how the code was already, but it seems simpler/idiomatic (and avoids an unwrap) to:
```rust
                     if let Some(timer) = res.request()
                                             .app_data::<AppState>()
                                             .expect("app data")
                                             .timer
                                             .replace(None)
                     {
                         timer.observe_duration();
                     }
```

File: components/sup/src/http_gateway.rs (hunk 1 above)
Comment (baumanj) [reply components/sup/src/http_gateway.rs]: I don't see a good way around it either.

**Pull request merged**